### DISCUSSION TITLE
Using Vue component as options for interface extension

### DISCUSSION THREAD

**Discussion by veryphatic** [directus/directus#11927] [Archive]
Hi team, the extensions docs for an interface **options** property suggests you can use either an options object or a dedicated Vue component https://docs.directus.io/extensions/interfaces/#extension-entrypoint

I want to use a Vue component, but when I pass a Vue component using the options API (i.e. Vue/ComponentOptions) it fails to mount and throws a type error  TypeError: r.standard is not iterable 

Is this the right way to use a Vue component instead of mapping out system options?

**Comment by rijkvanzanten** [comment]
What does your Vue component look like, and how did you assign it to the options? It sounds like the check that sees if the object is a Vue component or regular object fails somehow 🤔

**Reply by veryphatic** [reply]
Hi, I've tried both the options API and composition API components, and also the non-TS version i.e. without defineComponent, just exporting the options object.

```
// Composition API
<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
    name: "Test",
    setup(props) {
        console.log('created', props);
        return { setup: true }
    }
})
</script>
```
```
// Options API
<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
    name: "Test",
    mounted() {
        console.log('component mounted');
    }
})
</script>
```
And this is how I register it:

```
import { defineInterface } from '@directus/extensions-sdk';
import MyInterface from './MyInterfaceComponent.vue';
import Test from './test.vue';

export default defineInterface({
	id: 'veryphatic-interface',
	name: 'MyInterface ',
	icon: 'box',
	description: 'Example description',
	component: MyInterface ,
	types: ['alias'],
	localTypes: ['o2m'],
	group: 'relational',
	relational: true,
	options: Test
});
```

**Comment by veryphatic** [comment]
@rijkvanzanten i can also confirm the extension is being loaded when Directus starts

**Comment by veryphatic** [comment]
Anyone able to confirm if the way I'm mounting the options component correctly, or is this a bug?

**Comment by rijkvanzanten** [comment]
I haven't had the time to test it out myself yet, but could you open a new issue (bug report) and include an exact reproduction? Thanks!

**Comment by jonahweissman** [comment]
I didn't get the exact error as you, but I also had trouble figuring out how to use a Vue component for options. I ended up modeling my options component after [the options component in the list interface](https://github.com/directus/directus/blob/d81450235541f900b3c8937882cc3caad2d7ff5f/app/src/interfaces/list/options.vue). My problem was that directus doesn't seem to support options components that use `<script setup>`. Obviously that's not the problem with your code, but it might be helpful to see how the internal components use this feature.

**Reply by veryphatic** [reply]
Thanks for the bump. I found my issue was the component needs all properties of the single file component for the component to build correctly. That is, it needs the `template`, `script `and `style` properties. I'm also able to declare components using the `<script setup>` header without any issues. 

I also referred to the list component as a good example of using a component for options.

**Reply by jonahweissman** [reply]
Thanks for following up with what you learned! My component seems to work with no `style` property. Pretty mysterious, I would have assumed that Vue compiles all of these different formats to the same object.
